src/api/ApiCall.hh

```diff
@@ -0,0 +1,143 @@
+#ifndef RA_API_CALL_HH
+#define RA_API_CALL_HH
+#pragma once
+
+#include "services\IThreadPool.hh"
+#include "services\ServiceLocator.hh"
+
+#include <string>
+
+namespace ra {
+namespace api {
+
+enum class ApiResult
+{    
+    None = 0,       // unspecified
+    Success,        // call was successful
+    Error,          // an error occurred
+    Failed,         // call was unsuccessful, but didn't error
+    Unsupported,    // call is not supported and was not made
+};
+
+struct ApiRequestBase
+{
+protected:
+    ApiRequestBase() noexcept = default;
+    ~ApiRequestBase() noexcept = default;
+    ApiRequestBase(const ApiRequestBase&) = default;
+    ApiRequestBase& operator=(const ApiRequestBase&) = default;
+    ApiRequestBase(ApiRequestBase&&) noexcept = default;
+    ApiRequestBase& operator=(ApiRequestBase&&) noexcept = default;
+
+    template<class TRequest, class TCallback>
+    static void CallAsync(const TRequest& request, TCallback&& callback)
+    {
+        ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync([request, callback = std::move(callback)]
+            {
+                callback(request.Call());
+            });
+    }
+};
```

Comment:
`ApiRequestBase` provides a generic `CallAsync` that dispatches the specific `Call` method to a background thread. There are no common fields on `ApiRequestBase`